controllers/propagator/replicatedpolicy_controller.go

```diff
@@ -620,13 +622,131 @@ func (r *ReplicatedPolicyReconciler) cleanUpReplicated(ctx context.Context, repl
+// policy template. On retryable errors, the generated compliance event is added to the ComplianceServerCtx queue
+// for MonitorDatabaseConnection to handle once the database is back up.
+func (r *ReplicatedPolicyReconciler) recordDisabledEvents(
+	ctx context.Context, replicatedPolicy *policiesv1.Policy,
+) {
+	log := log.WithValues("policy", replicatedPolicy.Name)
+
+	if replicatedPolicy.Annotations[ParentPolicyIDAnnotation] == "" {
+		return
+	}
+
+	parentPolicyID, err := strconv.ParseInt(replicatedPolicy.Annotations[ParentPolicyIDAnnotation], 10, 32)
+	if err != nil {
+		log.Error(err, "Failed to record a disabled compliance event due to an invalid parent policy ID")
+
+		return
 	}
 
-	return deleteErr
+	toQueue := []*complianceeventsapi.EventDetailsQueued{}
+	dbConnectionDown := false
+
+	for _, template := range replicatedPolicy.Spec.PolicyTemplates {
+		plcTmplUnstruct := &unstructured.Unstructured{}
+
+		err := plcTmplUnstruct.UnmarshalJSON(template.ObjectDefinition.Raw)
+		if err != nil {
+			continue
+		}
+
+		policyIDStr := plcTmplUnstruct.GetAnnotations()[PolicyIDAnnotation]
+		if policyIDStr == "" {
+			continue
+		}
+
+		policyID, err := strconv.ParseInt(policyIDStr, 10, 32)
+		if err != nil {
+			log.Error(err, "Failed to record a disabled compliance event due to an invalid policy ID")
+
+			continue
+		}
+
+		complianceEvent := &complianceeventsapi.EventDetailsQueued{
+			ParentPolicyID: int32(parentPolicyID),
+			PolicyID:       int32(policyID),
+			Compliance:     "Disabled",
+			Message:        "The policy was removed because the parent policy no longer applies to this cluster",
+			Timestamp:      time.Now().UTC(),
+			ReportedBy:     "governance-policy-framework",
+		}
+
+		if dbConnectionDown {
+			log.Info(
+				"Failed to record the compliance event. Will requeue.",
+				"error", complianceeventsapi.ErrDBConnectionFailed.Error(),
+				"eventMessage", complianceEvent.Message,
+				"policyID", complianceEvent.PolicyID,
+			)
+
+			toQueue = append(toQueue, complianceEvent)
+
+			continue
+		}
+
+		var logSuffix string
+
+		err = complianceeventsapi.RecordLocalClusterComplianceEvent(
+			ctx, r.ComplianceServerCtx, complianceEvent.EventDetails(),
+		)
+		if errors.Is(err, complianceeventsapi.ErrRetryable) {
+			logSuffix = ". Will requeue."
+
+			toQueue = append(toQueue, complianceEvent)
+		}
+
+		if errors.Is(err, complianceeventsapi.ErrDBConnectionFailed) {
+			dbConnectionDown = true
+		}
+
+		if err != nil {
+			log.Info(
+				"Failed to record the compliance event"+logSuffix,
+				"error", err.Error(),
+				"eventMessage", complianceEvent.Message,
+				"policyID", complianceEvent.PolicyID,
+			)
+		} else {
+			log.V(2).Info(
+				"Recorded the compliance event",
+				"eventMessage", complianceEvent.Message,
+				"policyID", complianceEvent.PolicyID,
+			)
+		}
+	}
+
+	for _, event := range toQueue {
+		event, err := json.Marshal(event)
```

Comment:
workqueue uses a map under the hood so anything not hashable can't be used. It seems the Metadata field being JSONMap is what was tripping it up, so I'll readjust to not use JSON since that field isn't used here.